### DISCUSSION TITLE
Destruct Composable Store

### DISCUSSION THREAD

**Discussion by rynz** [vuejs/pinia#851] [Help and Questions] [answered]
```ts
export const useCounterStore = defineStore('counter', () => {
  const count = ref(0)
  function increment() {
    count.value++
  }

  return { count, increment }
})
```
Is very similar to `composables` which when destructed do not require `toRefs`. 

When exclusively using a Composable Store as per the example above, must we still use `storeToRefs`? Or does it behave exactly like a `composable`?

**Comment by posva** [comment, marked as answer]
You still ned `toRefs()` or `storeToRefs()` because it adds more than just what is returned and wraps it with `reactive()`

**Reply by rynz** [reply]
Thank you, good to know it wraps it with `reactive()`. Is it something you might consider changing for Composable Stores to mimic that of other features in reactivity?

**Reply by posva** [reply]
No, because this way you can simply do `store.aState` instead of `store.aState.value`. Passing through a custom function to destructure is more flexible. This could be implemented with a vite plugin maybe #718

**Reply by rynz** [reply]
Okay fair enough, personally I find `.value` outside of templates a great way to explicitly define what is reactive. I primarily use `ref` for this reason. However, I do understand your reasoning and appreciate the answer. I'll shy away from destructuring my store modules and rather reference the reactive properties.

Thank you for such an amazing project. I have been using pinia since alpha and it's been a joy to work with!
